File: angular-ui/src/app/orders/order-request.dto.ts
class OrderItemDto {
  id?: number

  constructor(data: Partial<OrderItemDto>) { Object.assign(this, data); }

}

export class OrderRequestDto {
  items?: OrderItemDto[];
  discountCode?: string;

  constructor(data: Partial<OrderRequestDto>) { Object.assign(this, data); }
}

File: backend/lms/src/main/java/com/el/order/application/dto/OrderRequestDTO.java
package com.el.order.application.dto;

import jakarta.validation.Valid;
import jakarta.validation.constraints.NotEmpty;

import java.util.Set;

public record OrderRequestDTO(
        @NotEmpty(message = "Items can't be empty")
        @Valid
        Set<OrderItemDTO> items,
        String discountCode
) {

}
